### PR TITLE
Add pre-submit job for reconciler to verify upgrading Kyma2 latest to Kyma2 main

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Add a pre-submit job (for reconciler image bumps, and changes to mothership/component reconciler charts) to verify upgrading Kyma2 latest to Kyma2 main on gardener.

**Related issue(s)**

- https://github.com/kyma-incubator/reconciler/issues/804

### CODE REVIEW

File: prow/jobs/control-plane/control-plane-reconciler-upgrade-kyma2-latest-to-main-gardener.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pre-main-control-plane-reconciler-upgrade-kyma2-latest-to-main"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-bot-github-token: "true"
+        preset-cluster-version: "true"
+        preset-gardener-gcp-kyma-integration: "true"
+        preset-kyma-cli-stable: "true"
+      run_if_changed: '^resources/kcp/values.yaml|^resources/kcp/charts/mothership-reconciler/|^resources/kcp/charts/component-reconcilers/'
+      optional: true
```
Comment (mfaizanse) [inline prow/jobs/control-plane/control-plane-reconciler-upgrade-kyma2-latest-to-main-gardener.yaml:19]: Should it be an optional job?

Comment (marcobebway) [reply prow/jobs/control-plane/control-plane-reconciler-upgrade-kyma2-latest-to-main-gardener.yaml:19]: As agreed with @k15r that we will keep this optional for now, until the reconciler becomes more stable.

**Review by mfaizanse — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**